packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/search/search_dsl/query_params.ts

```diff
@@ -323,6 +362,79 @@ const getMatchPhrasePrefixFields = ({
   return output;
 };
 
+const trimBoostAndWildcardFromField = (field: string) => {
+  return field.replace(/[*^].$/, '');
+};
+
+const getFieldsByQueryType = ({
+  searchFields,
+  types,
+  mappings,
+}: {
+  searchFields: string[];
+  types: string[];
+  mappings: IndexMapping;
+}): {
+  simpleQuerySearchFields: string[];
+  nestedQueryFields: Map<string, string[]>;
+} => {
+  const simpleQuerySearchFields: Set<string> = new Set();
+  const nestedQueryFields: Map<string, string[]> = new Map();
+
+  types.forEach((type) => {
+    searchFields.forEach((rawSearchField) => {
+      const searchField = trimBoostAndWildcardFromField(rawSearchField);
```

Comment:
I think we can remove `trimBoostAndWildcardFromField` entirely since we discard everything after the last `.` when we do `parentNode = absoluteFieldPath.split('.').slice(0, -1).join('.');`